### DISCUSSION TITLE
Conflicting code

### DISCUSSION THREAD

**Discussion by Pandagineer** [bitcraze/discussions#239] [Q&A]
Hi. I have a theoretical question. Let's say I write a *.c code which instructs the Crazyflie (and flowdeck) to hover at 1 m. I compile it and flash it. Then later, I write a second *.c code which instructs the Crazyflie to hover at 0.5 m. I compile it and flash it. What would happen? Thanks!

**Comment by knmcguire** [comment]
Hi!

Interesting question. I am assuming that you will be using the [app layer](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/userguides/app_layer/) for this right?

It depends on which level of autonomy you are sending this on, so see this documentation for the framework overview: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/userguides/app_layer/

In the firmware code, there are two ways to send a command to hover from within the crazyflie.

- [commanderSetSetpoint(&setpoint, priority_level)](https://github.com/bitcraze/crazyflie-firmware/blob/4e08de30256ea479db7c1186060a2d2920359bf2/src/modules/src/commander.c#L74)
- [high level commander command takeoff](https://github.com/bitcraze/crazyflie-firmware/blob/4e08de30256ea479db7c1186060a2d2920359bf2/src/modules/src/crtp_commander_high_level.c#L436) or [go_to](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/crtp_commander_high_level.c#L557).

The commandSetSetpoint is much more basic, where you have to set the full setpoint and[ the priority level.](https://github.com/bitcraze/crazyflie-firmware/blob/4e08de30256ea479db7c1186060a2d2920359bf2/src/modules/interface/commander.h#L42-L43) If you would be controlling it from the cflib, the priority for sending regular setpoints would be 2 and for the highlevel would be 1. Since higher number means higher priority, the regular commander setpoint set [will take over](https://github.com/bitcraze/crazyflie-firmware/blob/4e08de30256ea479db7c1186060a2d2920359bf2/src/modules/src/commander.c#L81).

In your case, it will be totally in your control if you use the first setSetpoint function. If you use an higher number in one part of the code than the other, that is probably pretty predictable as the setpoint with the highest priority will be used. If they are both on the same priority, that will be interesting though. I think that in that cae the Crazyflie will just jitter between the two setpoints as it flies between them.

**Reply by Pandagineer** [reply]
Thank you. The priority notion makes a lot of sense. (It was truly a curiosity question. I'm still learning... for example, I don't know what "app layer" means. I'll likely come back to this thread in 1 year to review.)

BTW, is it possible to hire Bitcraze to consult if I have specific questions or goals? Thanks again.

**Reply by knmcguire** [reply]
We don't offer such services ourselves, but we always help out here on discussions or by email if something can't be discussed online.  

If you are beginning out, perhaps you could also check out the youtube channel. This recorded lecture is good for those who are starting out: https://youtu.be/6s8i-nhPjt0
